src/Inventory/Asset/Software.php

```diff
@@ -525,7 +525,7 @@ protected function getFullCompareKey(\stdClass $val, bool $with_version = true):
             'name'             => mb_strtolower($val->name),
             'version'          => $with_version ? mb_strtolower($val->version ?? '') : '',
             'arch'             => mb_strtolower($val->arch ?? ''),
-            'manufacturers_id' => mb_strtolower(transliterator_transliterate("Any-Latin; Latin-ASCII; [^a-zA-Z0-9\.\ -_] Remove;", $val->manufacturers_id)),
+            'manufacturers_id' => mb_strtolower(transliterator_transliterate("Any-Latin; Latin-ASCII; [^a-zA-Z0-9\.\ -_] Remove;", $val->manufacturers_id ?? '')),
```

Comment:
mmented [now](https://github.com/glpi-project/glpi/pull/18523#issuecomment-2535002196)
@Lainow

Can you ask for the inventory file that causes this error (I don't think I've ever seen software without a manufacturer).

$val->manufacturers_id should always exist, if that's not the case, it's the prepare function that needs to be fixed.